logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/FileSharingStatus.kt

```diff
@@ -18,4 +18,10 @@
 
 package com.wire.kalium.logic.configuration
 
-data class FileSharingStatus(val isFileSharingEnabled: Boolean?, val isStatusChanged: Boolean?)
+data class FileSharingStatus(val state: FileSharingState, val isStatusChanged: Boolean?)
+
+sealed interface FileSharingState {
+    object Disabled : FileSharingState
+    object EnabledAll : FileSharingState
+    data class EnabledSome(val allowedType: List<String>) : FileSharingState
+}
```

Comment:
The naming is just too similar to `FileSharingStatus`.

Perhaps moving it so it is an inner class?
`FileSharingStatus.Value`